tests/test_TFNetworkLayer.py

```diff
@@ -146,6 +146,131 @@ def test_batch_norm_vars():
+    seq_lens = np.array([11, 11, 5, 11, 11, 11, 11, 11, 11, 11], dtype=np.float32)
+    n1 = 9 * 11 * 16 + 5 * 16
+    n2 = 10 * 11 * 16
+
+    with tf.variable_scope("batch_norm_masked_nchw"):
+      batch_norm_1 = BatchNormLayer(name="batch_norm_masked_nchw", network=net, masked_time=True,
+                                    sample_mean=mean, sample_variance=variance,
+                                    use_shift=False, use_std=False, epsilon=0.0,
+                                    sources=[src_nhwc],
+                                    output=BatchNormLayer.get_out_data_from_opts(name="batch_norm_masked_nchw",
+                                                                                 sources=[src_nhwc],
+                                                                                 network=net))
+      batch_norm_1.post_init(layer_desc=None)
+    with tf.variable_scope("batch_norm_nonmasked_nchw"):
+      batch_norm_2 = BatchNormLayer(name="batch_norm_nonmasked_nchw", network=net, masked_time=False,
+                                    sample_mean=mean, sample_variance=variance,
+                                    use_shift=False, use_std=False, epsilon=0,
+                                    sources=[src_nhwc],
+                                    output=BatchNormLayer.get_out_data_from_opts(name="batch_norm_nonmasked_nchw",
+                                                                                 sources=[src_nhwc],
+                                                                                 network=net))
+      batch_norm_2.post_init(layer_desc=None)
+    tf.global_variables_initializer().run()
+    out_1, seq_lens_1 = session.run([batch_norm_1.output.placeholder,
+                                     batch_norm_1.output.size_placeholder[0]],
+                                    feed_dict={src_nhwc.output.placeholder: input_data,
+                                               src_nhwc.output.size_placeholder[0]: seq_lens}
+                                    )
+    out_2, seq_lens_2 = session.run([batch_norm_2.output.placeholder,
+                                 batch_norm_2.output.size_placeholder[0]],
+                                feed_dict={src_nhwc.output.placeholder: input_data_masked,
+                                           src_nhwc.output.size_placeholder[0]: seq_lens}
+                                )
+    # Manually calculating batch_norm and compare to the tf output
+    np_bn2 = (input_data - data_mean) * (1.0 / np.sqrt(data_var))
+    npt.assert_array_almost_equal(np_bn2, out_2, decimal=5)
+    # Manually calculating batch_norm with different seq_lens, having:
+    # Mean_1 = n2 / n1 * Mean_2
+    # Var_1 = n2 / n1 * (Var_2 + Mean_2 ^ 2 (1 - n2 / n1))
+    # bn_1 = (x - Mean_1) * 1 / sqrt(Var_1)
+    # Substituting Mean_1 and Var_1:
+    np_bn1 = (input_data - n2 / n1 * data_mean) * \
+             (1.0 / np.sqrt(n2 / n1 * (data_var + data_mean ** 2 * (1 - n2 / n1))))
```

Comment:
`n2`/`n1` is an `int`, so this behaves different in Pyhton 2/3.
You maybe should add `from __future__ import division` in the beginning of the file, then they are always the same.